[Darren] iP

Darren12345677 opened the pull request:
# Roofus
> “Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes).

Roofus frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download the latest release from [here](https://github.com/Darren12345677/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Roofus(System.getProperty("user.home") 
            + "/data/roofus.txt").run(); 
}
```

Commit:
Add Gradle support

Darren12345677 committed:
Level-1

Darren12345677 committed:
Add, List

Darren12345677 committed:
Mark as done

Darren12345677 committed:
ToDos, Events, Deadlines

Darren12345677 committed:
Revert "ToDos, Events, Deadlines"

This reverts commit d06a604811a9a4c09961cd758a1da1a496db6087.

Darren12345677 committed:
ToDos, Events, Deadlines

Darren12345677 committed:
Test using the I/O redirection technique

Darren12345677 committed:
Follow naming conventions

Darren12345677 committed:
change if-else to switch case

Darren12345677 committed:
Revert "change if-else to switch case"

This reverts commit ed0ea94172b9afb56c400275e7b8e35ef4665eac.

Darren12345677 committed:
change if-else to switch case

Darren12345677 committed:
Handle Errors

Darren12345677 committed:
Delete

Darren12345677 committed:
Use Enums

Darren12345677 committed:
Use Exceptions to handle errors

created RoofusException class to represent exceptions specific to Roofus

Darren12345677 committed:
Enable save feature

load() method to transfer data from storage to task list in Roofus
save() method to save data from task list in Roofus to storage
load() is called when program starts
save() is called when program terminates

Darren12345677 committed:
Add date and time

Darren12345677 committed:
Merge branch 'branch-Level-7'

Darren12345677 committed:
Merge branch 'branch-Level-8'

Darren12345677 committed:
Change format of saved data

Darren12345677 committed:
Create necessary directories for file path

Darren12345677 committed:
Add more OOP

Darren12345677 committed:
Adjust test cases

Darren12345677 committed:
Divide class into packages

Darren12345677 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Darren12345677 committed:
Add JUnit tests

Darren12345677 committed:
Configure build.gradle

Darren12345677 committed:
Add JavaDoc comments

Darren12345677 committed:
Tweak the code to comply with a coding standard

Darren12345677 committed:
Give users a way to find a task by searching for a keyword

Darren12345677 committed:
Merge branch 'Level-9'

Darren12345677 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/roofus/Ui.java

Darren12345677 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/roofus/Parser.java
#	src/main/java/roofus/TaskList.java
#	src/main/java/roofus/Ui.java
#	src/main/java/roofus/command/Command.java
#	src/main/java/roofus/task/Deadline.java

Darren12345677 committed:
Update tests for find command

Darren12345677 committed:
Add JavaDoc comments

Darren12345677 committed:
Use CheckStyle to detect coding style violations

Darren12345677 committed:
Use JavaFX technology in tutorial

Darren12345677 committed:
Add a GUI to Roofus

Darren12345677 committed:
Merge branch 'branch-Level-10'

Darren12345677 committed:
Adjust GUI

Darren12345677 committed:
Adjust tasklist to save on change

maxtance commented on src/main/java/roofus/task/ToDo.java:
```diff
@@ -0,0 +1,34 @@
+     * {@inheritDoc}
+     */
+    @Override
+    public String writeString() {
+        return String.format("T | %d | %s",
+                super.isDone ? 1 : 0, super.description);
+    }
```
The layout for this method is done well and is tidy! Correct number of indentations and good use of space character.

maxtance commented on src/main/java/roofus/command/Command.java:
```diff
@@ -0,0 +1,29 @@
+     * Checks if Roofus should still be running after command is executed
+     *
+     * @return boolean Boolean representing whether current instance of
+     *     Roofus should terminate.
```
This is not clear to me why the boolean represents whether the Roofus object should terminate. Might want to consider "returns true if Roofus is currently running; false otherwise".

maxtance commented on src/main/java/roofus/Ui.java:
```diff
@@ -0,0 +1,129 @@
+
+    /**
+     * Formats and prints Roofus's greetings.
+     */
```
Could include @return in the comments.

maxtance commented on src/main/java/roofus/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+     * Constructs an instance of TaskList.
+     *
+     * @param taskString List of strings to be saved as Tasks.
+     * @see Task
```
First time learning about the @see tag. Good use of tag 👍

maxtance commented on src/main/java/roofus/Parser.java:
```diff
@@ -0,0 +1,106 @@
+                }
+                return new FindCommand(split[1]);
+            default:
+                break;
```
I like that the switch block is neat and tidy. The indentation for the switch and case is done correctly. Well done 👍

maxtance reviewed (commented):
Overall very neat code. Implemented javadocs for almost all class and method headers which is impressive! Well done 😄

Darren12345677 committed:
Update JavaFX modules in build.gradle

Roofus has a GUI which is implemented by JavaFx technology

Adding the necessary JavaFX modules to build.gradle enables
Gradle to run Roofus

Darren12345677 committed:
Use Assertions to define assumptions about program state

Commands classes have methods with pre-conditions on the parameters
and the commands cannot be executed if these conditions are not met.

Using Assertions to ensure that the index the command should act on is
valid.

Let's assert that indices are within the boundaries of the task list.

Darren12345677 committed:
Improve code quality

Darren12345677 committed:
Merge pull request #1 from Darren12345677/branch-A-Assertions

Use Assertions to define assumptions about program state

Darren12345677 committed:
Merge branch 'master' of https://github.com/Darren12345677/ip

Darren12345677 committed:
Merge branch 'master' into branch-A-CodeQuality

Darren12345677 committed:
Merge pull request #2 from Darren12345677/branch-A-CodeQuality

Improve code quality

Darren12345677 committed:
Merge branch 'master' of https://github.com/Darren12345677/ip into branch-A-CodeQuality

Darren12345677 committed:
Merge branch 'master' of https://github.com/Darren12345677/ip

Darren12345677 committed:
Minor bug fixes on assertions

Darren12345677 committed:
Implement comparable in Tasks

Darren12345677 committed:
Add command to sort tasks chronologically

Darren12345677 committed:
Merge pull request #3 from Darren12345677/branch-C-Sort

Add command to sort tasks chronologically

Darren12345677 committed:
Merge branch 'master' of https://github.com/Darren12345677/ip

Darren12345677 committed:
Merge branch 'branch-A-CodeQuality'

Darren12345677 committed:
Improve the GUI

Darren12345677 committed:
Give a unique personality

Darren12345677 committed:
Handle more errors

Darren12345677 committed:
Upload Ui screenshot

Darren12345677 committed:
Include more checks on date format

Darren12345677 committed:
Fix bug in dialog box

Darren12345677 committed:
Add a User Guide

Darren12345677 committed:
Update feature descriptions

Darren12345677 committed:
Fix bug when there is no storage file

Darren12345677 committed:
Add test for Ui

Darren12345677 committed:
Adjust JavaDocs

Darren12345677 committed:
Fix bug in Find command formatting

Darren12345677 committed:
Update javafx version

Darren12345677 committed:
Fix bug in delete command
